weblate/utils/tasks.py

```diff
@@ -83,20 +83,42 @@ def database_backup():
         return
     ensure_backup_dir()
     database = settings.DATABASES["default"]
-    if database["ENGINE"] != "django.db.backends.postgresql":
-        return
-    cmd = ["pg_dump", "--dbname", database["NAME"]]
-    if database["HOST"]:
-        cmd += ["--host", database["HOST"]]
-    if database["PORT"]:
-        cmd += ["--port", database["PORT"]]
-    if database["USER"]:
-        cmd += ["--username", database["USER"]]
-    if settings.DATABASE_BACKUP == "compressed":
-        cmd += ["--file", data_dir("backups", "database.sql.gz")]
-        cmd += ["--compress", "6"]
+    if database["ENGINE"] == "django.db.backends.postgresql":
+        cmd = ["pg_dump", "--dbname", database["NAME"]]
+        if database["HOST"]:
+            cmd += ["--host", database["HOST"]]
+        if database["PORT"]:
+            cmd += ["--port", database["PORT"]]
+        if database["USER"]:
+            cmd += ["--username", database["USER"]]
+        if settings.DATABASE_BACKUP == "compressed":
+            cmd += ["--file", data_dir("backups", "database.sql.gz")]
+            cmd += ["--compress", "6"]
+        else:
+            cmd += ["--file", data_dir("backups", "database.sql")]
+    elif database["ENGINE"] == "django.db.backends.mysql":
+        options = data_dir("home") + "/.my.cnf"
+
+        with open(options, "w") as handle:
+            handle.write("[mysqldump]\n")
+            handle.write("password = '" + database["PASSOWRD"] + "'\n")
```

Comment:
I think using MYSQL_PWD environment variable is a better approach than creating/updating a file with configuration on every run. When creating the file, you would have to deal with non-existing directories as well (this is what fails in the CI).

Reply:
Funny how that's not in the MySQL dump man page.  I was trying to avoid it, but at least it would be consistent with Postgresql.

Was reading this.

https://dev.mysql.com/doc/refman/8.0/en/password-security-user.html

BTW, impressed by your prechecks/CI.  Really cool stuff.

Reply:
Consistency with PostgreSQL is not really needed in this, I just think it's less error prone to pass it in environment instead of creating file to hold it.